src/components/DirectPaymentCreateModal/DirectPaymentCreateModal.tsx

```diff
@@ -286,142 +309,265 @@ export const DirectPaymentCreateModal: React.FC<DirectPaymentCreateModalProps> =
     });
   };
 
+  const renderConfirmationScreen = () => {
+    if (!paymentDataToConfirm) return null;
+
+    // Helper function to shorten wallet addresses
+    const shortenAddress = (
+      address: string,
+      startChars: number = 6,
+      endChars: number = 8,
+    ): string => {
+      if (!address || address.length <= startChars + endChars + 3) return address;
+      return `${address.slice(0, startChars)}...${address.slice(-endChars)}`;
+    };
```

Comment:
We already have a helper `shortenAccountKey.ts`. We should update it if we want it to be more flexible and use it here.